concordium-contracts-common/src/schema.rs

```diff
@@ -704,83 +711,62 @@ impl Deserial for FunctionV1 {
 
 impl Serial for FunctionV2 {
     fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
-        match self {
-            FunctionV2::Param(parameter) => {
-                out.write_u8(0)?;
-                parameter.serial(out)
-            }
-            FunctionV2::Rv(return_value) => {
-                out.write_u8(1)?;
-                return_value.serial(out)
-            }
-            FunctionV2::ParamRv {
-                parameter,
-                return_value,
-            } => {
-                out.write_u8(2)?;
-                parameter.serial(out)?;
-                return_value.serial(out)
-            }
-            FunctionV2::Error(error) => {
-                out.write_u8(3)?;
-                error.serial(out)
-            }
-            FunctionV2::ParamError {
-                parameter,
-                error,
-            } => {
-                out.write_u8(4)?;
-                parameter.serial(out)?;
-                error.serial(out)
-            }
-            FunctionV2::RvError {
-                return_value,
-                error,
-            } => {
-                out.write_u8(5)?;
-                return_value.serial(out)?;
-                error.serial(out)
-            }
-            FunctionV2::ParamRvError {
-                parameter,
-                return_value,
-                error,
-            } => {
-                out.write_u8(6)?;
-                parameter.serial(out)?;
-                return_value.serial(out)?;
-                error.serial(out)
-            }
+        let parameter = self.parameter.is_some();
+        let return_value = self.return_value.is_some();
+        let error = self.error.is_some();
+        let tag: u8 = match (parameter, return_value, error) {
+            // parameter
+            (true, false, false) => 0,
+            // return_value
+            (false, true, false) => 1,
+            // parameter + return_value
+            (true, true, false) => 2,
+            // error
+            (false, false, true) => 3,
+            // parameter + error
+            (true, false, true) => 4,
+            // return_value + error
+            (false, true, true) => 5,
+            // parameter + return_value + error
+            (true, true, true) => 6,
+            // no schema: cannot happen
+            (false, false, false) => 7,
```

Comment:
If this case cannot happen, then consider returning an error instead

Reply:
I updated the comment:

` // no schema:`

Reply:
The type, with three `Option`s, technically allow this case. But, hopefully, our tools should never construct a `FunctionSchemaV2` with three `None`s. 
We could make the fields private and add a `new` method to check that at least one of the schemas was supplied. But I don't see much benefit in that because it is only used internally and because we probably access the fields directly in several places.
So let's just keep the current version.